cli/Sources/TuistServer/Client/ServerAuthenticationController.swift

```diff
@@ -74,92 +88,175 @@ public struct ServerAuthenticationController: ServerAuthenticationControlling {
         #endif
     }
 
+    public func refreshToken(serverURL: URL, inSubprocess: Bool) async throws {
+        try await authenticationTokenRefreshingIfNeeded(
+            serverURL: serverURL,
+            forceRefresh: true,
+            inSubprocess: inSubprocess
+        )
+    }
+
     public func refreshToken(serverURL: URL) async throws {
         try await authenticationTokenRefreshingIfNeeded(
-            serverURL: serverURL, forceRefresh: true
+            serverURL: serverURL,
+            forceRefresh: true,
+            inSubprocess: true
         )
     }
 
     @discardableResult private func authenticationTokenRefreshingIfNeeded(
         serverURL: URL,
-        forceRefresh: Bool
+        forceRefresh: Bool,
+        inSubprocess: Bool
     ) async throws
         -> AuthenticationToken?
     {
         #if canImport(TuistSupport)
             Logger.current.debug("Refreshing authentication token for \(serverURL) if needed")
         #endif
-        return
-            try await CachedValueStore.current
-                .getValue(key: "token_\(serverURL.absoluteString)") {
-                    () -> (value: AuthenticationToken, expiresAt: Date?)? in
-                    guard let token = try await fetchTokenFromStore(serverURL: serverURL) else {
-                        return nil
-                    }
+        switch (try await tokenStatus(serverURL: serverURL, forceRefresh: forceRefresh), inSubprocess) {
+        case let (.valid(token), _):
+            return token
+        case (.absent, _), (_, false):
+            return try await executeRefresh(serverURL: serverURL, forceRefresh: forceRefresh)?.value
```

Comment:
how can we execute a refresh with an absent token?

Reply:
The function throws if the token is absent, but I'll hoist it here to make the workflow more obvious.